### PR TITLE
add 'post' report, include basic tests

### PR DESCRIPTION
**Pull request by jh-bate**
@krystophv - here are the updates we spoke about 

1. Allow `get` and `post` with common code for the blip vs internal service call uses-cases 
2. Update for consistency to use `require` - it was a mix of `import` and `require` which the tests complained about
3. Very basic unit tests for report utility functions. I kept these un-exported but have tested using `rewire`

### CODE REVIEW

File: lib/handlers/index.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const getUserReportsHandler = require('./getUserReportsHandler');
+const postUserReportsHandler = require('./postUserReportsHandler');
+const userDataHandler = require('./userDataHandler');
+
+module.exports = {
```
Comment (jh-bate) [inline lib/handlers/index.js:5]: just moved the handlers into a common handlers dir and export from index

File: lib/handlers/postUserReportsHandler.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  return async (req, res) => {
+    // Set the timeout for the request. Make it 10 seconds longer than
+    // our configured timeout to give the service time to cancel the API data
+    // request, and close the outgoing data stream cleanly.
+    req.setTimeout(exportTimeout + 10000);
+
+    try {
+      const userId = req.params.userid;
+      // TODO: which token will we use here?
+      const restrictedToken = '';
```
Comment (jh-bate) [inline lib/handlers/postUserReportsHandler.js:31]: TBD based on discussions

File: lib/reportUtils.js [hunk 3]
```diff
@@ -391,33 +405,16 @@ function getDateRangeByReport(
   return dates;
 }
 
-/**
- *
- * @param {object} res http response
- * @param {object} pdf pdf report data to send
- */
-export async function sendPDFReport(res, pdf) {
```
Comment (jh-bate) [inline lib/reportUtils.js:399]: now just in each handler

File: lib/reportUtils.js [hunk 4]
```diff
@@ -562,3 +559,117 @@ export async function processAGPSVGs(agpPDFData) {
...
+ *   userId: string;
+ *   fullName: string;
+ *   dob: string;
+ *   mrn: string;
+ *  }} userDetail
+ * @param {{
+ *   tzName: string;
+ *   bgUnits: string;
+ *   reports: Array;
+ *   dates: ReportDates;
```
Comment (jh-bate) [inline lib/reportUtils.js:584]: added dates so these can be passed through - I still need to use these dates

File: lib/handlers/postUserReportsHandler.js (hunk 2 above)
Comment (krystophv) [reply lib/handlers/postUserReportsHandler.js:31]: For the `POST` endpoint, the token will come in on the `x-tidepool-session-token` header as usual, regardless of whether it's from a user session or a server token, so we don't need to worry about the `restrictedToken` here.

**Pull request merged**